assemble.go

```diff
@@ -118,7 +135,30 @@ func AssembleFile(ctx context.Context, name string, idx Index, s Store, seeds []
 					ss.add(job.segment)
 					continue
 				}
+
+				// If we don't have a seedSegment we expect an IndexSegment with just
+				// a single chunk, that we can take from either the selfSeed, from the
+				// destination file, or from the store.
+				if len(job.segment.chunks()) != 1 {
+					panic("Received an unexpected segment that doesn't contain just a single chunk")
+				}
 				c := job.segment.chunks()[0]
+
+				// If we already took this chunk from the store we can reuse it by looking
+				// into the selfSeed.
+				if segment := ss.getChunk(c.ID); segment != nil {
+					copied, cloned, err := segment.WriteInto(f, c.Start, c.Size, blocksize, isBlank)
+					if err != nil {
+						return err
+					}
+					stats.addBytesCopied(copied)
+					stats.addBytesCloned(cloned)
+					// Record this segment's been written in the self-seed to make it
+					// available going forward
+					ss.add(job.segment)
```

Comment:
Interesting, this wouldn't even be necessary anymore, since we just confirmed we have the chunk in the self-seed already and it's all operating on a single chunk. It won't hurt of course, would only be useful if there was a whole segment we wanted to take from the selfseed. But it might be good to keep it and avoid any surprises when the logic is changed to support whole sections of self-seed. Should we update the comment to say it's not actually needed?

Reply:
Good point. I removed the `ss.add()` and wrote a comment to explain why it is not needed.
In this way, if in the future we add support for whole segments, it will be clear enough that the assumption of a single chunk would not be valid anymore.